src/test/java/redis/clients/jedis/tests/JedisTest.java

```diff
@@ -190,14 +179,40 @@ public void allowUrlWithNoDBAndNoPassword() {
     }
   }
 
+  @Test
+  public void uriWithDBindexShouldUseTimeout() throws URISyntaxException {
+    URI uri = new URI("redis://fakehost:6378/1");
+    long startTime = System.nanoTime();
+    try (Jedis j = new Jedis(uri, 5000)) {
+      j.ping();
+    } catch (Exception ex) {
+      assertEquals(JedisConnectionException.class, ex.getClass());
+      assertEquals(java.net.UnknownHostException.class, ex.getCause().getClass());
+    }
+    long stopTime = System.nanoTime();
+    assertTrue(stopTime - startTime > 4000);
```

Comment:
I don't get this one, the timeout is 5 seconds, but we are validating the runtime is greater than 4 **milli**seconds?

Reply:
Socket connections are dependent on OS. We can't strictly check the socket timeouts. I'm using the threshold to reduce this test's probability to become another [flaky](https://github.com/redis/jedis/issues/2367) test.

Reply:
Got you, so that would make it "less" flaky, but potentially still flaky, do we want to avoid that check completely? We are kind of checking the JDK's Socket timeout works as expected here which is not really adding much, I think just checking the exception is enough in that case

Reply:
I really want to have test for #1824 as this is a reported issue. And I can't find any other way to test it.

Reply:
This should do the trick? (a bit more involved, but should be a lot more stable)

```suggestion
    int port = 6677;
    int timeoutMillis = 5000;
    int deltaMillis = 500;

    URI uri = new URI("redis://localhost:" + port + "/1");
    ServerSocket server = new ServerSocket(port);
    Instant start = Instant.now();
    try (Jedis jedis = new Jedis(uri, timeoutMillis)) {
        jedis.ping();
    } catch (JedisConnectionException ex) {
        assertEquals(SocketTimeoutException.class, ex.getCause().getClass());
        assertEquals(timeoutMillis, Duration.between(start, Instant.now()).toMillis(), deltaMillis);
        return;
    } finally {
        server.close();
    }
    fail();
```

Reply:
I'll give that a try.